herd/semExtra.ml

```diff
@@ -497,6 +499,34 @@ type concrete =
 
     type pp_barrier = { barrier:barrier ; pp:string; }
 
+  let patch_variant test variant cache_type =
+    let dic_pred, idc_pred =
+        let open CacheType in
+          match cache_type with
+          | None ->
+              (fun _ -> false), (fun _ -> false)
+          | Some cache_type ->
+              cache_type.dic, cache_type.idc in
+    let procs =
+      List.map (fun (p,_,_) -> p) (test.Test_herd.start_points) in
+    let all_dic, all_idc, any_dic, any_idc = List.fold_left
+        (fun (_all_dic,_all_idc,_any_dic,_any_idc) p -> (
```

Comment:
Out of curiosity, why did you put an underscore at the beginning of those names? Usually I do that when I want to remove a "unused variable" warning, but you are using those after

Reply:
No good reason at all, it could be anything instead. (But it won't, in line with your next comment. :) )